frontend/src/contexts/authStateContext.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useKeycloak } from '@react-keycloak/web';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import { useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+
+export interface IAuthState {
+  authenticated?: boolean;
+  userInfo?: any;
+  ready?: boolean;
+}
+
+export const AuthStateContext = React.createContext<IAuthState>({
+  authenticated: false,
+  ready: false,
+});
+
+export const AuthStateContextProvider = (props: { children?: any }) => {
+  const keycloak = useKeycloakWrapper();
+  const [userInfo, setUserInfo] = React.useState<any>(null);
+  const keyCloakReady: boolean = useSelector<RootState, boolean>(state => state.keyCloakReady);
+
+  React.useEffect(() => {
+    const loadUserInfo = async () => {
```

Comment:
I like the addition of keycloak readiness tracking in this provider. I'm not sure if I see the benefit of tracking the userInfo here as well as in the keycloak provider. Is the intent for this provider to track all of the keycloak data instead of the existing keycloak provider?

Reply:
We don't have to track user info. But I realized that the useKeycloakWrapper does not always have the roles because the userInfo is not loaded. Just something to think about.

Reply:
I don't think its a problem to track it in more places, just wanted to make sure I understand.